pkgs/development/python-modules/gpt4all-bindings/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  gpt4all,
+  gpt4all-cuda,
+  buildPythonPackage,
+  stdenv,
+  requests,
+  tqdm,
+  pythonOlder,
+  setuptools,
+  wheel,
+  python,
+  cuda ? false,
+}:
+let
+  basepkg = if cuda then gpt4all-cuda else gpt4all;
+
+  gpt4all-backend = stdenv.mkDerivation {
+    pname = "gpt4all-backend";
+    inherit (basepkg)
+      src
+      version
+      cmakeFlags
+      nativeBuildInputs
+      buildInputs
+      ;
+
+    sourceRoot = "${basepkg.src.name}/gpt4all-backend";
+    installPhase = ''
+      cp -r ./ $out
+    '';
+  };
+in
+buildPythonPackage {
+  pname = "gpt4all-bindings";
+  inherit (basepkg) src version;
+  sourceRoot = "${basepkg.src.name}/gpt4all-bindings/python";
+  pyproject = false;
+
+  patches = [ ./setup_fixup.patch ];
+  disabled = pythonOlder "3.12";
+
+  propagatedBuildInputs = [
+    requests
+    tqdm
+  ];
+
+  build-system = [
+    setuptools
+    wheel
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail 'SRC_CLIB_DIRECTORY = ' 'SRC_CLIB_DIRECTORY = "${gpt4all-backend}" #' \
+      --replace-fail 'SRC_CLIB_BUILD_DIRECTORY = ' 'SRC_CLIB_BUILD_DIRECTORY = "${gpt4all-backend}/" #' \
+      --replace-fail 'DEST_CLIB_BUILD_DIRECTORY = ' 'DEST_CLIB_BUILD_DIRECTORY = os.path.join(DEST_CLIB_DIRECTORY, "build") #'
+  '';
+
+  buildPhase = ''
+    python3 setup.py build
+  '';
+
+  installPhase = ''
+    mkdir -p $out/${python.sitePackages}/
+    cp -r gpt4all $out/${python.sitePackages}/gpt4all
+  '';
+
+  pythonImportsCheck = [ "gpt4all" ];
+
```

Comment:
Could you enable tests?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

Reply:
So I enabled tests, and after removing those needing network connection to download a large LLM and do inference, it left me with 0 tests left to execute.
So I just let them disabled, and added a comment on why